winterfell/services/claimService.js

```diff
@@ -2,18 +2,88 @@ var _ = require('lodash');
 var http = require('http-status-codes');
 var httpErrors = require('./../utils/httpErrors');
 var Claim = require('./../models/claim');
+var Form = require('./../models/form');
+var Q = require('q');
+var bulk = require('bulk-require');
+var formlyFields = bulk(__dirname + '/../forms/', ['*']);
+var expressions = require("angular-expressions");
 
 function ClaimService(app) {
   this.app = app;
 }
 
+/**
+ * For a given form and set of responses, calculate how many questions
+ * were answered (which is simply the number of keys in the responses),
+ * and more complicatedly: how many questions on the form were answerable,
+ * which requires evaluating the angular hideExpression attribute for
+ * each field against the current responses.
+ *
+ * @param formName
+ * @param data
+ * @returns {{answerable: number, answered: number}}
+ */
+function calculateProgress(formName, data) {
+  var evaluate, i;
+  var template = formlyFields[formName];
```

Comment:
we want to be careful with what `formName` vs. `formId` means. I think there might be some inconsistencies in a few places (I've done it as well).

To me,
`formId` = `VBA-21-0966-ARE1`
`formRowId` = mongoDb row id of form
`formName` = "INTENT TO FILE A CLAIM FOR COMPENSATION AND/OR PENSION, OR SURVIVORS PENSION AND/OR DIC" (as seen in `VBA-21-0966-ARE.json`) We could also call this formTitle

`formUnofficialName` = "Intent to File" (our easier-to-read form title). We could also call this `formUnofficialTitle` or maybe something like `form.vfi.name` to convey that everything in the object `vfi` is our stuff.
Thoughts?